frontend/connection/src/ForwardMessageCache.ts

```diff
@@ -100,32 +125,22 @@ export class ForwardMsgCache {
       return msg
     }
 
-    let newMsg = this.getCachedMessage(msg.refHash as string, true)
+    const newMsg = this.getCachedMessage(msg.refHash as string, true)
     if (notNullOrUndefined(newMsg)) {
       LOG.info(`Cached ForwardMsg HIT [hash=${msg.refHash}]`)
     } else {
-      // Cache miss: fetch from the server
-      LOG.info(`Cached ForwardMsg MISS [hash=${msg.refHash}]`)
-      const encodedNewMsg = await this.endpoints.fetchCachedForwardMsg(
-        msg.refHash as string
+      throw new Error(
+        `Cached ForwardMsg MISS [hash=${msg.refHash}]. This is not expected to happen. Please [report this bug](https://github.com/streamlit/streamlit/issues).`
       )
-      try {
-        newMsg = ForwardMsg.decode(encodedNewMsg)
-      } catch (e) {
-        throw new Error(
-          `Failed to decode ForwardMsg (hash=${msg.refHash}): ${
-            (e instanceof Error ? e : new Error(`${e}`)).message
-          }`
-        )
-      }
-
-      this.maybeCacheMessage(newMsg, encodedNewMsg)
```

Comment:
At which point do we cache the message now if not here?

Reply:
There is another `maybeCacheMessage` at the top of `processMessagePayload`. The removed `maybeCacheMessage` was used to cache message received from the cached message endpoint.